src/services/codefixes/fixImportNonExportedMember.ts

```diff
@@ -0,0 +1,229 @@
+            resolvedModule.resolvedFileName
+        );
+
+        if (!originSourceFile) {
+            return undefined;
+        }
+
+        return { node, importDecl, originSourceFile };
+    }
+
+    function getNamedExportDeclaration(
+        sourceFile: SourceFile
+    ): ExportDeclaration | undefined {
+        let namedExport;
+
+        for (const statement of sourceFile.statements) {
+            if (
+                isExportDeclaration(statement) &&
+                statement.exportClause &&
+                isNamedExports(statement.exportClause)
+            ) {
+                namedExport = statement;
+            }
+        }
+
+        return namedExport;
+    }
+
+    function compareIdentifiers(s1: Identifier, s2: Identifier) {
+        return compareStringsCaseInsensitive(s1.text, s2.text);
+    }
+
+    function sortSpecifiers(
+        specifiers: ExportSpecifier[]
+    ): readonly ExportSpecifier[] {
+        return stableSort(specifiers, (s1, s2) =>
+            compareIdentifiers(
+                s1.propertyName || s1.name,
+                s2.propertyName || s2.name
+            )
+        );
+    }
+
+    const isVariableDeclarationListWith1Element = (
+        node: Node
+    ): node is VariableDeclarationList =>
+        !(isVariableDeclarationList(node) && node.declarations.length !== 1);
+
+    function doChange(
+        changes: textChanges.ChangeTracker,
+        sourceFile: SourceFile,
+        node: Identifier
+    ): void {
+        const moduleSymbol = sourceFile.localSymbol ?? sourceFile.symbol;
+
+        const localSymbol = moduleSymbol.valueDeclaration?.locals?.get(
+            node.escapedText
+        );
+
+        if (localSymbol === undefined) {
+            return;
+        }
+
+        // Node we need to export is a function
+        if (isFunctionSymbol(localSymbol)) {
+            const node = localSymbol.valueDeclaration;
+
+            if (node === undefined) {
+                return;
+            }
+
+            if (!isDeclarationStatement(node) && !isVariableStatement(node)) {
+                return;
+            }
+
+            return changes.insertExportModifier(sourceFile, node);
+        }
+        // Node we need to export is a variable declaration
+        else if (
+            localSymbol.valueDeclaration &&
+            isVariableDeclarationListWith1Element(
+                localSymbol.valueDeclaration.parent
+            )
+        ) {
+            const node = localSymbol.valueDeclaration.parent;
+
+            return changes.insertExportModifier(sourceFile, node);
+        }
+
+        // In all other cases => the node is a variable, and should be exported via `export {a}`
+        const namedExportDeclaration = getNamedExportDeclaration(sourceFile);
+
+        // If there is an existing export
+        if (
+            namedExportDeclaration?.exportClause &&
+            isNamedExports(namedExportDeclaration.exportClause)

```

Comment:
Should we check if this is a type-only export? My understanding is that this would change:

```ts
export type { existingThing };
```

to

```ts
export { existingThing, newThing };
```

which may not be desired.

Reply:
nice! yes, this is a  good test case, adding now